stepfunctions/src/main/scala/awscala/stepfunctions/StepFunctions.scala

```diff
@@ -0,0 +1,131 @@
+    this.setRegion(region)
+    this
+  }
+
+  def createStateMachine(name: String, definition: String, roleArn: String): StateMachine = StateMachine {
+    createStateMachine(
+      new CreateStateMachineRequest()
+        .withName(name)
+        .withDefinition(definition)
+        .withRoleArn(roleArn)).getStateMachineArn
+  }
+
+  def stateMachine(name: String): Option[StateMachine] = stateMachines.find(_.name == name)
+
+  def stateMachines: Seq[StateMachine] = {
+    object StateMachineSequencer extends Sequencer[StateMachineListItem, ListStateMachinesResult, String] {
+      val base = new ListStateMachinesRequest()
+      def getInitial: ListStateMachinesResult = listStateMachines(base)
+
+      def getMarker(r: ListStateMachinesResult): String = r.getNextToken
+
+      def getFromMarker(marker: String): ListStateMachinesResult = listStateMachines(base.withNextToken(marker))
+
+      def getList(r: ListStateMachinesResult): util.List[StateMachineListItem] = r.getStateMachines
+    }
+    StateMachineSequencer.sequence.map(sm => StateMachine(sm.getStateMachineArn))
+  }
+
+  def createActivity(name: String): Activity = Activity {
+    createActivity(new CreateActivityRequest().withName(name)).getActivityArn
+  }
+
+  def activity(name: String): Option[Activity] = activities.find(_.name == name)
+
+  def activities: Seq[Activity] = {
+    object ActivitiesSequencer extends Sequencer[ActivityListItem, ListActivitiesResult, String] {
+      private val base = new ListActivitiesRequest()
+      def getInitial: ListActivitiesResult = listActivities(base)
+
+      def getMarker(r: ListActivitiesResult): String = r.getNextToken
+
+      def getFromMarker(marker: String): ListActivitiesResult = listActivities(base.withNextToken(marker))
+
+      def getList(r: ListActivitiesResult): util.List[ActivityListItem] = r.getActivities
+    }
+    ActivitiesSequencer.sequence.map(a => Activity(a.getActivityArn))
+  }
+
+  def runActivity(name: String, workerName: String = null)(fn: String => String)(
+    implicit
+    ec: ExecutionContext): Future[Option[String]] =
+    Future {
+      activity(name).map(_.arn).map { arn =>
+        blocking {
+          getActivityTask(new GetActivityTaskRequest().withActivityArn(arn).withWorkerName(workerName))
+        }
+      }
+    }.map { s =>
+      s.flatMap(ac => Option(ac.getTaskToken).map((_, ac.getInput)))
+        .map {
+          case (token, input) =>
+            val result = Try(fn(input))
+            result.fold(
+              err =>
+                sendTaskFailure(
+                  new SendTaskFailureRequest()
+                    .withTaskToken(token)
+                    .withCause(Option(err.getCause).map(_.getMessage).orNull)
+                    .withError(err.getMessage)),
+              output => sendTaskSuccess(new SendTaskSuccessRequest().withTaskToken(token).withOutput(output)))
+            result.get
+        }
+    }
+
+}
+
+class StepFunctionsClient(
```

Comment:
Have another source file for this class.

Reply:
No problem breaking this out, but like my earlier comment all the other modules have the client classes within their "main" class. Should I follow that pattern?